internal/resources/cluster/backupschedule/resource_schema.go

```diff
@@ -0,0 +1,321 @@
+package backupschedule
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+const (
+	ResourceName = "tanzu-mission-control_backup_schedule"
+
+	// Root Keys
+	nameKey        = "name"
+	clusterNameKey = "cluster_name"
+	specKey        = "spec"
+	// provisionerNameKey       = "provisioner_name"
+	// managementClusterNameKey = "management_cluster_name"
+
+	// Spec Directive Keys
+	pausedKey   = "paused"
+	scheduleKey = "schedule"
+	templateKey = "template"
+
+	// Schedule Directive Keys
+	rateKey = "rate"
+
+	// Template Directive Keys
+	backupTTLKey                = "backup_ttl"
+	excludedNamespacesKey       = "excluded_namespaces"
+	includedNamespacesKey       = "included_namespaces"
+	excludedResourcesKey        = "excluded_resources"
+	includedResourcesKey        = "included_resources"
+	includeClusterResourcesKey  = "include_cluster_resources"
+	defaultVolumesToResticKey   = "default_volumes_to_restic"
+	snapshotVolumesKey          = "snapshot_volumes"
+	csiSnapshotTimeoutKey       = "csi_snapshot_timeout"
+	defaultVolumesToFsBackupKey = "default_volumes_to_fs_backup"
+	storageLocationKey          = "storage_location"
+	volumeSnapshotLocationsKey  = "volume_snapshot_locations"
+	orderedResourcesKey         = "ordered_resources"
+	hooksKey                    = "hooks"
+	labelSelectorKey            = "label_selector"
+	orLabelSelectorKey          = "or_label_selector"
+
+	// Hooks Directive Keys
+	resourceKey = "resource"
+
+	// Resource Directive Keys
+	preHookKey  = "pre_hook"
+	postHookKey = "post_hook"
+
+	// Pre/Post Hook Directive Keys
+	execKey = "exec"
+
+	// Exec Directive Keys
+	commandKey   = "command"
+	containerKey = "container"
+	onErrorKey   = "on_error"
+	timeoutKey   = "timeout"
+
+	// (Or)Label Selector Directive Keys
+	matchLabelsKey     = "match_labels"
+	matchExrpessionKey = "match_expression"
+
+	// Match Expressions Directive Keys
+	meKey         = "key"
+	meOperatorKey = "operator"
+	meValuesKey   = "values"
+)
+
+var backupScheduleResourceSchema = map[string]*schema.Schema{
+	nameKey:        nameSchema,
+	clusterNameKey: clusterNameSchema,
+	specKey:        specSchema,
+	common.MetaKey: common.Meta,
```

Comment:
I see the data protection resource in [this](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/265) PR also has `managementClusterNameKey` and `provisionerNameKey`. Is that not needed here?

Reply:
I am not sure if it is required in the API itself, I will finish testing today and adjust

Reply:
All fixed